pkg/controllers/deprovisioning/types.go

```diff
@@ -121,19 +121,31 @@ func (o Command) String() string {
 			fmt.Fprintf(&buf, "/%s", capacityType)
 		}
 	}
-	if o.replacementNode != nil {
-		ct := o.replacementNode.Requirements.Get(v1alpha5.LabelCapacityType)
+	odNodes := 0
+	spotNodes := 0
+	for _, node := range o.replacementNodes {
+		ct := node.Requirements.Get(v1alpha5.LabelCapacityType)
+		if ct.Has(v1alpha5.CapacityTypeOnDemand) {
+			odNodes++
+		}
+		if ct.Has(v1alpha5.CapacityTypeSpot) {
+			spotNodes++
+		}
+	}
+	// Print list of instance types for the first replacementNode.
+	if len(o.replacementNodes) > 1 {
+		fmt.Fprintf(&buf, " and replacing with %d spot and %d on-demand nodes from types %s",
+			spotNodes, odNodes,
+			scheduling.InstanceTypeList(o.replacementNodes[0].InstanceTypeOptions))
```

Comment:
nit: Couldn't this instance type option list get really long?

Reply:
No, the InstanceTypeList function truncates it with "and %d others"

Reply:
Nope, the InstanceTypeList code just outputs the first five instance types. Check here https://github.com/aws/karpenter-core/blob/main/pkg/controllers/provisioning/scheduling/node.go#L125-L138